### PR TITLE
update aws sdk to 1.0.x

### PR DESCRIPTION
**Pull request by ivaxer**
## Problem

AWS Rust SDK is generally available now: https://aws.amazon.com/blogs/developer/announcing-general-availability-of-the-aws-sdk-for-rust/

This change will be useful for experimenting with S3 performance.

### CODE REVIEW

File: s3_scrubber/src/checks.rs [hunk 1]
```diff
@@ -307,7 +304,9 @@ pub(crate) async fn list_timeline_blobs(
         errors.push("S3 list response got no index_part.json file".to_string());
     }
 
-    if let Some(index_part_object_key) = index_part_object.as_ref().and_then(|object| object.key())
+    if let Some(index_part_object_key) = index_part_object
+        .as_ref()
+        .and_then(|object| Some(object.key()))
```
Comment (ivaxer) [inline s3_scrubber/src/checks.rs:309]: fixed it

Comment (koivunej) [reply s3_scrubber/src/checks.rs:309]: Was this a test failure? Your comment predates the test report.

File: libs/remote_storage/src/s3_bucket.rs [hunk 2]
```diff
@@ -98,18 +99,20 @@ impl S3Bucket {
             .set_max_attempts(Some(1))
             .set_mode(Some(RetryMode::Adaptive));
 
-        let mut config_builder = Config::builder()
+        let mut config_builder = Builder::default()
+            .behavior_version(BehaviorVersion::v2023_11_09())
```
Comment (koivunej) [inline libs/remote_storage/src/s3_bucket.rs:103]: Need this over at s3_scrubber/lib.rs

Comment (koivunej) [reply libs/remote_storage/src/s3_bucket.rs:103]: [437c8ec](https://github.com/neondatabase/neon/pull/5976/commits/437c8eca5ccc030f66202afba44e38ef563e4f20)

File: s3_scrubber/src/checks.rs (hunk 1 above)
Comment (koivunej) [reply s3_scrubber/src/checks.rs:309]: Ok perhaps this was just an API change?

**Review by koivunej — APPROVED** [review]
Assuming tests pass, this looks good to merge, I will check the test results tomorrow. Thanks for helping with this!

File: Cargo.lock [hunk 3]
```diff
@@ -6030,12 +6206,13 @@ dependencies = [
...
  "regex-syntax 0.7.2",
  "reqwest",
- "ring",
+ "ring 0.16.20",
  "rustls",
  "scopeguard",
  "serde",
  "serde_json",
  "smallvec",
+ "subtle",
```
Comment (koivunej) [inline Cargo.lock:6215]: TODO(me): check what our duplicate deps look like now since this shuffles a lot of those.

**Comment by koivunej** [comment]
Handling that clippy before leaving...

File: s3_scrubber/src/checks.rs (hunk 1 above)
Comment (koivunej) [reply s3_scrubber/src/checks.rs]: Later: [aab4a34](https://github.com/neondatabase/neon/pull/5976/commits/aab4a346136f16c59d8523c29c43c132d1083907)

**Comment by koivunej** [comment]
> #### Failures on Postgres 14
> 
>     * `test_delete_timeline_exercise_crash_safety_failpoints[Check.RETRY_WITH_RESTART-real_s3-timeline-delete-before-index-deleted-at]`: [release](https://neon-github-public-dev.s3.amazonaws.com/reports/pr-5978/7036363656/index.html#suites/f588e0a787c49e67b29490359c589fae/d4df4ec4cd3184a9)

Looked like basic real_s3 flakyness.

File: Cargo.lock (hunk 3 above)
Comment (koivunej) [reply Cargo.lock:6215]: Impressive: aws-sigv4 brings two versions of crypto-bigint.

Other than it looks like our only old ring dependency is jsonwebtoken which has gone through a major bump, possibly for ring... I think that's all new duplicates.

Comment (ivaxer) [reply Cargo.lock:6215]: @koivunej, Are we going to fix it in this PR?

Comment (koivunej) [reply Cargo.lock:6215]: No no, this was just for myself (and internally for anyone interested in duplicate deps and why we cannot update them).

**Pull request merged**